docs/udp.rst

```diff
@@ -34,79 +34,188 @@ This turns your openEO process into a web application
-            "arguments": {"x": {"from_node": "subtract32"}, "y": 1.8},
-            "result": true
+.. code-block:: json
+
+      {
+        "process_id": "load_collection",
+        "arguments": {
+          "bands": [
+            "B02",
+            "B04",
+            "B08",
+            "SCL"
+          ],
+          "id": "SENTINEL2_L2A",
+          "spatial_extent": {
+            "from_parameter": "bbox"
+          },
+          "temporal_extent": {
+            "from_parameter": "temporal_interval"
+          }
         }
-    }
+      }
 
 
-The important point here is the parameter reference ``{"from_parameter": "fahrenheit"}`` in the subtraction.
-When we call this user-defined process we will have to provide a Fahrenheit value.
-For example with 70 degrees Fahrenheit (again in openEO JSON format here)::
+The important point here is the parameter reference ``{"from_parameter": "bbox"}``
+and ``{"from_parameter": "temporal_interval"}`` in the above process graph.
+When we call this UDP, we will have to provide a value as shown below:
 
-    {
-        "process_id": "fahrenheit_to_celsius",
-        "arguments" {"fahrenheit": 70}
-    }
+.. code-block:: json
 
+    {
+        "process_id": "EVI_timeseries",
+        "arguments": {
+          "temporal_interval": [
+            "2022-09-01T00:00:00Z",
+            "2022-10-01T00:00:00Z"
+          ],
+          "bbox": {
+            "type": "Polygon",
+            "coordinates": [
+              [
+                ...
+              ]
+            ]
+          }
+        },
+        "result": true
+      }
 
 Declaring Parameters
 ---------------------
 
-It's good style to declare what parameters your user-defined process expects and supports.
+It's a good style to declare what parameters your user-defined process expects and supports.
 It allows you to document your parameters, define the data type(s) you expect
 (the "schema" in openEO-speak) and define default values.
 
 The openEO Python client lets you define parameters as
 :class:`~openeo.api.process.Parameter` instances.
-In general you have to specify at least the parameter name,
+In general, you have to specify at least the parameter name,
 a description and a schema (to declare the expected parameter type).
-The "fahrenheit" parameter from the example above can be defined like this::
 
-    from openeo.api.process import Parameter
+The ``bbox`` and the ``temporal_interval`` parameters from the above example can be defined like this:
+
+.. code-block:: python
 
-    fahrenheit_param = Parameter(
-        name="fahrenheit",
-        description="Degrees Fahrenheit",
-        schema={"type": "number"}
+    temporal_interval = Parameter(
+        name = "temporal_interval",
+        description = "The date range to load.",
+        schema = {"type": "array", "subtype": "temporal-interval"},
+        default = ["2018-06-15", "2018-06-27"]
+    )
+    polygon = Parameter(
+        name = "bbox",
```

Comment:
as mentioned above, it is confusing to link "polygon" (which sounds very geojson-like) with "bbox" (which is just a west-east-south-north thing)